lib/utils.js

```diff
@@ -81,4 +81,35 @@ class NullLogger {
 }
 utils.nullLogger = new NullLogger();
 
+/**
+ * Deep merge two objects.
+ * @param {Object} target
+ * @param {Object} ...sources
+ * @return {Object}
+ */
+utils.mergeDeep = (target, ...sources) => {
+    if (!sources.length) {
+        return target;
+    }
+    const source = sources.shift();
```

Comment:
It's actually better/faster to iterate over all of the sources rather than modifying the array on every pop.

Reply:
done